src/did/src/send_raw_transaction.rs

```diff
@@ -0,0 +1,486 @@
+            }),
+            TRANSACTION_TYPE_EIP2930 => TransactionData::Eip2930(Eip2930 {
+                chain_id: value.chain_id.ok_or("Chain id is missing")?,
+                nonce: value.nonce,
+                gas_price: value.gas_price.ok_or("Gas price is missing")?,
+                gas_limit: value.gas,
+                to: value.to,
+                value: value.value,
+                access_list: value.access_list.ok_or("Access list is missing")?,
+                input: value.input.into(),
+            }),
+            _ => return Err("Unknown transaction type"),
+        };
+
+        Ok(SendRawTransactionRequest {
+            signature,
+            transaction: data,
+        })
+    }
+}
+
+/// Transaction type and data
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize)]
+pub enum TransactionData {
+    Legacy(Legacy),
+    Eip2930(Eip2930),
+    Eip1559(Eip1559),
+}
+
+/// Transaction input
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
+pub enum TransactionInput {
+    #[default]
+    Transfer,
+    Call(Bytes),
+}
+
+impl From<Bytes> for TransactionInput {
+    fn from(value: Bytes) -> Self {
+        if value.0.is_empty() {
+            return TransactionInput::Transfer;
+        }
+        TransactionInput::Call(value)
+    }
+}
+
+impl From<TransactionInput> for Bytes {
+    fn from(value: TransactionInput) -> Self {
+        match value {
+            TransactionInput::Transfer => Bytes::default(),
+            TransactionInput::Call(bytes) => bytes,
+        }
+    }
+}
+
+/// Legacy transaction format
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
+pub struct Legacy {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// Added as EIP-155: Simple replay attack protection
+    pub chain_id: Option<U256>,
+    /// A scalar value equal to the number of transactions sent by the sender; formally Tn.
+    pub nonce: U256,
+    /// A scalar value equal to the number of
+    /// Wei to be paid per unit of gas for all computation
+    /// costs incurred as a result of the execution of this transaction; formally Tp.
+    ///
+    /// As ethereum circulation is around 120mil eth as of 2022 that is around
+    /// 120000000000000000000000000 wei we are safe to use u128 as its max number is:
+    /// 340282366920938463463374607431768211455
+    pub gas_price: U256,
+    /// A scalar value equal to the maximum
+    /// amount of gas that should be used in executing
+    /// this transaction. This is paid up-front, before any
+    /// computation is done and may not be increased
+    /// later; formally Tg.
+    pub gas_limit: U256,
+    /// The 160-bit address of the message call’s recipient or, for a contract creation
+    /// transaction, ∅, used here to denote the only member of B0 ; formally Tt.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub to: Option<H160>,
```

Comment:
This should be the `TxKind` mentioned above. Check the `TxKind` implementation of alloy